storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -1333,6 +1336,21 @@ private AssignmentDistributionService getAssignmentsDistributer() {
         return heartbeatsCache;
     }
 
+    public AtomicReference<Map<String, Set<List<Integer>>>> getIdToExecutors() {
+        return idToExecutors;
+    }
+
+    private Set<List<Integer>> getOrUpdateExecutors(String topoId, StormBase base, Map<String, Object> topoConf,
+                                                    StormTopology topology)
+        throws IOException, AuthorizationException, InvalidTopologyException, KeyNotFoundException {
+        Set<List<Integer>> executors = idToExecutors.get().get(topoId);
+        if (null == executors) {
+            executors = new HashSet<>(computeExecutors(topoId, base, topoConf, topology));
+            idToExecutors.getAndUpdate(new Assoc<>(topoId, executors));
```

Comment:
Do we intend to create new HashMap every time? If not, would `idToExecutors.get().put(topoId, executors);` should suffice?